### PR TITLE
Take advantage of clang-formats new rules

### PR DESCRIPTION
**Pull request by TrentHouliston**
Because why not

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -4,51 +4,93 @@ Language: Cpp
...
-AllowShortBlocksOnASingleLine: false
+AllowAllArgumentsOnNextLine: true
+AllowAllConstructorInitializersOnNextLine: true
+AllowAllParametersOfDeclarationOnNextLine: true
+AllowShortBlocksOnASingleLine: true
 AllowShortCaseLabelsOnASingleLine: true
 AllowShortFunctionsOnASingleLine: Empty
-AllowShortIfStatementsOnASingleLine: true
+AllowShortLambdasOnASingleLine: All
+AllowShortIfStatementsOnASingleLine: WithoutElse
```
[the remark this answers is not in the document]
Comment (TrentHouliston) [reply .clang-format]: I figured that would be the most divisive. It should only apply for very short if statements that don't have else statements that fit on a single line? Or is that the thing you don't like.

[the remark this answers is not in the document]
Comment (JosephusPaye) [reply .clang-format]: Vote for multi-line always.

File: .clang-format [hunk 2]
```diff
@@ -4,51 +4,93 @@ Language: Cpp
...
 AlignConsecutiveDeclarations: false
-AlignEscapedNewlinesLeft: false
+AlignEscapedNewlines: Left
 AlignOperands: true
 AlignTrailingComments: true
+AllowAllArgumentsOnNextLine: false
+AllowAllConstructorInitializersOnNextLine: true
 AllowAllParametersOfDeclarationOnNextLine: false
-AllowShortBlocksOnASingleLine: false
+AllowShortBlocksOnASingleLine: true
```
Comment (JosephusPaye) [inline .clang-format]: Isn't `AllowShortBlocksOnASingleLine` this the same as `AllowShortIfStatementsOnASingleLine`, but for other blocks like `while`?

Would it be more consistent to configure them the same?

Comment (TrentHouliston) [reply .clang-format]: Probably

**Comment by ysims** [comment]
Should this be good to merge now?

**Comment by TrentHouliston** [comment]
Would need to be applied to the codebase first, but so long as everyone likes the style it's fine

**Pull request merged**